### PR TITLE
[APO-1838] Add TypeScript codegen support for SlackTrigger workflows

### PR DESCRIPTION
**Pull request by alex-nork**
The purpose of this PR is to add TypeScript codegen support for SlackTrigger workflows, including trigger type recognition, trigger attribute references (e.g., `SlackTrigger.message`), and serialization.

## Key Changes
- Added `SLACK_MESSAGE` trigger type mapping to generate `SlackTrigger` class references
- Implemented `TriggerAttributeWorkflowReference` class for generating AST nodes that reference trigger attributes
- Added `TriggerAttributeWorkflowReferenceSerializer` with proper snake_case conversion
- Updated `WorkflowValueDescriptorReference` union to include `TRIGGER_ATTRIBUTE` type
- Added Python serialization support for `TriggerAttributeReference` objects in `expressions.py`

## Test Coverage
- TypeScript: Graph generation, trigger attribute references, edge cases with snapshots
- Python: Workflow execution, serialization, trigger detection, attribute references
- All tests passing ✅

## Related
- Builds on APO-1833 (SlackTrigger base implementation)
- Part of Slack integration workflow support

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by alex-nork** [comment]
@codex review

File: ee/vellum_ee/workflows/display/utils/expressions.py [hunk 1]
```diff
@@ -347,6 +348,17 @@ def serialize_value(executable_id: UUID, display_context: "WorkflowDisplayContex
...
 
+    if isinstance(value, TriggerAttributeReference):
+        # Generate trigger ID using the same hash formula as in base_workflow_display.py
+        trigger_class = value.trigger_class
+        trigger_id = uuid4_from_hash(f"{trigger_class.__module__} | {trigger_class.__qualname__}")
+
+        return {
+            "type": "TRIGGER_ATTRIBUTE",
+            "triggerId": str(trigger_id),
+            "attributeId": str(value.id),
```
[the remark this answers is not in the document]
Comment (alex-nork) [reply ee/vellum_ee/workflows/display/utils/expressions.py]: @codex address this feedback

File: ee/codegen/src/__test__/graph-attribute.test.ts [hunk 2]
```diff
@@ -1101,5 +1101,59 @@ describe("Workflow", () => {
...
+
+      const graphAttribute = new GraphAttribute({
+        workflowContext,
+      });
+
+      graphAttribute.write(writer);
+
+      expect(writer.toString()).toMatchInlineSnapshot(
+        `"SlackTrigger >> FirstNode >> SecondNode"`
+      );
```
Comment (dvargasfuertes) [inline ee/codegen/src/__test__/graph-attribute.test.ts:1156]: [nit] pref towards file snapshots since they can be easily updated with `npm run test:update` when warranted

File: ee/codegen/src/generators/workflow-value-descriptor-reference/trigger-attribute-workflow-reference.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  private getTriggerClassInfo(
+    triggerType: string
+  ): { className: string; modulePath: string[] } | null {
+    switch (triggerType) {
+      case "MANUAL":
+        return {
+          className: "ManualTrigger",
+          modulePath: ["vellum", "workflows", "triggers", "manual"],
```
Comment (dvargasfuertes) [inline ee/codegen/src/generators/workflow-value-descriptor-reference/trigger-attribute-workflow-reference.ts:50]: [nit] we had a constant from a previous PR for the `["vellum", "workflows", "triggers"` module path that we could reuse here and below

File: ee/codegen/src/generators/graph-attribute.ts [hunk 4]
```diff
@@ -350,6 +350,11 @@ export class GraphAttribute extends AstNode {
           className: "ManualTrigger",
           modulePath: ["vellum", "workflows", "triggers", "manual"],
         };
+      case "SLACK_MESSAGE":
```
Comment (dvargasfuertes) [inline ee/codegen/src/generators/graph-attribute.ts:353]: [nit] seems like we could consolidate this method with our other switch method

File: ee/codegen/src/types/vellum.ts [hunk 5]
```diff
@@ -920,6 +926,7 @@ export type WorkflowValueDescriptorReference =
   | VellumSecretWorkflowReference
   | EnvironmentVariableWorkflowReference
   | ExecutionCounterWorkflowReference
+  | TriggerAttributeWorkflowReference
```
Comment (dvargasfuertes) [inline ee/codegen/src/types/vellum.ts:929]: [oos] we will need to introduce this reference on the vellum django side as well

File: ee/vellum_ee/workflows/display/utils/expressions.py [hunk 6]
```diff
@@ -347,6 +348,17 @@ def serialize_value(executable_id: UUID, display_context: "WorkflowDisplayContex
             "node_id": str(node_class_display.node_id),
         }
 
+    if isinstance(value, TriggerAttributeReference):
+        # Generate trigger ID using the same hash formula as in base_workflow_display.py
+        trigger_class = value.trigger_class
+        trigger_id = uuid4_from_hash(f"{trigger_class.__module__} | {trigger_class.__qualname__}")
```
Comment (dvargasfuertes) [inline ee/vellum_ee/workflows/display/utils/expressions.py:354]: [nit] most of our other hash's have no space on either side of the `|`

**Review by dvargas92495 — APPROVED** [review]
so 🔥
